docs/en/sql-reference/statements/check-grant.md

```diff
@@ -0,0 +1,43 @@
+---
+slug: /en/sql-reference/statements/check-grant
+sidebar_position: 56
+sidebar_label: CHECK GRANT
+title: "CHECK GRANT Statement"
+---
+
+The `CHECK GRANT` query is used to check whether the current user/role has been granted a specific privilege, and whether the corresponding table/column exists in the memory.
```

Comment:
@Unalian Why do you think `CHECK GRANT` should check that the corresponding table/column exists in the memory?

Reply:
> @Unalian Why do you think `CHECK GRANT` should check that the corresponding table/column exists in the memory?

<img width="964" alt="image" src="https://github.com/user-attachments/assets/47ad1fd9-5cbd-4ab3-8ab9-bd96166b42f5">
Depends on the describe of the issue.

Reply:
For example, if the table doesn't exist then
```
DROP TABLE IF EXISTS db.mytable`
```
still works, and it requires privilege `GRANT DROP TABLE ON db.mytable`. But what is the point of checking
```
CHECK GRANT DROP TABLE ON db.mytable
```
in this case?

Reply:
Perhaps we need either a setting to enable/disable checking existence in `CHECK GRANT` or just a separate command to check existence only. I mean we have `EXISTS TABLE` command, so I think we can have `EXISTS COLUMNS` as well.

Reply:
> For example, if a table doesn't exist then
> 
> ```
> DROP TABLE IF EXISTS db.mytable
> ```
> 
> still works, and that requires privilege `GRANT DROP TABLE ON db.mytable`.
> 
> But what is the point of checking
> 
> ```
> CHECK GRANT DROP TABLE ON db.mytable
> ```
> 
> in this case?

```SQL
CHECK GRANT DROP TABLE ON db.mytable
```
will return if the user has privilege to  ```DROP TABLE ON db.mytable``` and the ```db.mytable``` exists.

Reply:
> CHECK GRANT DROP TABLE ON db.mytable
will return if the user has privilege to DROP TABLE ON db.mytable and the db.mytable exists

Yes, but this is not what some users might expect from command `CHECK GRANT DROP TABLE ON db.mytable` in this case.

Reply:
> Perhaps we need either a setting to enable/disable checking existence in `CHECK GRANT` or separate commands to check grants only or to check existence only. We already have the `EXISTS TABLE` command, so I think we can have `EXISTS COLUMNS` as well.

I got your point.  Separating the commands to check grants only or to check existence only may be better. Actually, that is what I am thinking during coding(https://github.com/ClickHouse/ClickHouse/issues/67772#issuecomment-2288083101). I will change the logic here (remove the existence check in check grant). Then, maybe add ```EXISTS COLUMNS``` just like ```EXISTS TABLE`` as well.